src/app/shared/services/task/task.service.spec.ts

```diff
@@ -1,16 +1,207 @@
+      imports: [HttpClientTestingModule],
+      providers: [
+        {
+          provide: TemplateFieldService,
+          useValue: mockTemplateFieldService,
+        },
+        {
+          provide: AppDataService,
+          useValue: new MockAppDataService(MOCK_DATA),
+        },
+        {
+          provide: AppConfigService,
+          useValue: new MockAppConfigService(MOCK_CONFIG),
+        },
+        // Mock single method from campaign service called
+        {
+          provide: CampaignService,
+          useValue: {
+            ready: async () => {
+              return true;
+            },
+            scheduleCampaignNotifications: scheduleCampaignNotificationsSpy,
+          },
+        },
+      ],
+    });
     service = TestBed.inject(TaskService);
   });
 
   it("should be created", () => {
     expect(service).toBeTruthy();
   });
+  it("enables service via app config", async () => {
+    await service.ready();
+    expect(service.tasksFeatureEnabled).toEqual(true);
+  });
+  it("can get task group data rows", async () => {
+    await service.ready();
+    expect(await service.getTaskGroupDataRows(taskGroupsListName)).toEqual(
+      MOCK_DATA.data_list[taskGroupsListName].rows
+    );
+  });
+  it("can get highlighted task group (set to highest priority task group on init)", async () => {
+    await service.ready();
+    expect(service.getHighlightedTaskGroup()).toBe(
+      MOCK_DATA.data_list[taskGroupsListName].rows[0].id
+    );
+  });
+  it("can get highlighted task group index", async () => {
+    await service.ready();
+    expect(await service.getHighlightedTaskGroupIndex(taskGroupsListName)).toBe(0);
+  });
+  it("evaluates highlighted task group correctly after init", async () => {
+    await service.ready();
+    expect(service.evaluateHighlightedTaskGroup().previousHighlightedTaskGroup).toBe(
+      MOCK_DATA.data_list[taskGroupsListName].rows[0].id
+    );
+    expect(service.evaluateHighlightedTaskGroup().newHighlightedTaskGroup).toBe(
+      MOCK_DATA.data_list[taskGroupsListName].rows[0].id
+    );
+  });
+  it("checking whether a task group is highlighted returns the correct value", async () => {
+    await service.ready();
+    expect(
+      service.checkHighlightedTaskGroup(MOCK_DATA.data_list[taskGroupsListName].rows[1].id)
+    ).toBe(false);
+    expect(
+      service.checkHighlightedTaskGroup(MOCK_DATA.data_list[taskGroupsListName].rows[0].id)
+    ).toBe(true);
+  });
+  it("can set a task group's completed status", async () => {
+    await service.ready();
+    await service.setTaskGroupCompletedStatus(
+      MOCK_DATA.data_list[taskGroupsListName].rows[0].completed_field,
+      true
+    );
+    expect(
+      await mockTemplateFieldService.getField(
+        MOCK_DATA.data_list[taskGroupsListName].rows[0].completed_field
+      )
+    ).toBe(true);
+  });
+  it("completing the highlighted task causes the next highest priority task to be highlighted upon re-evaluation", async () => {
```

Comment:
nit(non-blocking) - for future dev, would it make sense to automatically re-evaluate after setting completed?